app/controllers/api_controller/report_results.rb

```diff
@@ -0,0 +1,24 @@
+class ApiController
+  module ReportResults
+    #
+    # Report Results Supporting Methods
+    #
+    def results_query_resource(object)
+      object.send("miq_report_results")
+    end
+
+    def show_reports
+      return show_generic(:reports) unless @req[:subcollection]
```

Comment:
@abellotti not totally happy with this as there is some duplication here that might be hard to extract, and it does involve passing a block through more than one method call, but it does address the usability issue of showing the result set on a `show` but hiding it on an `index`. @abellotti any thoughts?